unity-renderer/Assets/DCLPlugins/UUIDEventComponentsPlugin/UUIDComponent/OnPointerEventColliders.cs

```diff
@@ -137,10 +157,48 @@ void DestroyColliders()
                 Collider collider = colliders[i];
 
                 if (collider != null)
-                    UnityEngine.Object.Destroy(collider.gameObject);
+                    UnityEngine.Object.Destroy(collider);
             }
 
             colliders = null;
         }
+        
+        private void AddColliderName(Collider collider)
+        {
+            if (collider is MeshCollider meshCollider)
+                colliderNames[collider] = meshCollider.sharedMesh.name;
+            else
+                colliderNames[collider] = collider.gameObject.name;
+        }
+        
+        private void GenerateColliders(IDCLEntity entity)
+        {
+            var renderers = entity?.meshesInfo?.renderers;
+            if(renderers == null || renderers.Length == 0)
+                return;
+                
+            colliders = new MeshCollider[renderers.Length];
+
+            for (int i = 0; i < colliders.Length; i++)
+            {
+                if (renderers[i] == null)
+                    continue;
+                
+                GameObject onPointerEventGameObjectCollider = new GameObject(COLLIDER_NAME);
+
+                onPointerEventGameObjectCollider.name = COLLIDER_NAME;
+                onPointerEventGameObjectCollider.layer = PhysicsLayers.onPointerEventLayer; 
+                onPointerEventGameObjectCollider.transform.SetParent(renderers[i].transform, false);
+                
+                // We need to increase it a little bit since if the shape is pointer blocker, it shouldn't block its own pointer event
+                // This way, there is no collision fighting 
+                onPointerEventGameObjectCollider.transform.localScale *= 1.0001f;
```

Comment:
This looks hacky =S, I think we should check how we are dealing with this situation (having the BoxShape in NFTshapes for example) in the current ECS6 implementation and do the same

Reply:
The current ECS6 implementation is bugged with this. Since the BoxShape in the NFTShapes never blocks the pointer event.  With the current implementation of ECS6, this is the only way to fix it ( in both ECS). 

In my opinion, we should use only 1 collider that acts as everything

Reply:
This has been removed with the current approach